spec/features/posts_index_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'rails_helper'
+
+RSpec.describe 'posts index', type: :feature do
+  let!(:user1) do
+    User.create(
+      name: 'test user1',
+      photo: 'https://images3.alphacoders.com/690/690494.jpg',
+      bio: 'test bio1',
+      posts_counter: 1
+    )
+  end
+
+  let!(:post1) do
+    Post.create(author: user1, title: 'Post 1', text: 'Post 1 content', comments_counter: 1, likes_counter: 0)
+  end
+  let!(:post2) do
+    Post.create(author: user1, title: 'Post 2', text: 'Post 2 content', comments_counter: 1, likes_counter: 0)
+  end
+
+  let!(:comment1) do
+    Comment.create(post: post1, user: user1, text: 'nice!')
+  end
+  let!(:comment2) do
+    Comment.create(post: post1, user: user1, text: 'nice!')
+  end
+  let!(:comment3) do
+    Comment.create(post: post1, user: user1, text: 'nice!')
+  end
+  let!(:comment4) do
+    Comment.create(post: post1, user: user1, text: 'nice!')
+  end
+  let!(:comment5) do
+    Comment.create(post: post1, user: user1, text: 'nice!')
+  end
+
+  describe 'Posts content' do
+    before(:example) do
+      visit user_posts_path(user1)
+    end
+
+    it 'displays the user\'s profile picture' do
+      expect(page).to have_selector("img[src='#{user1.photo}']")
+    end
+    it 'displays the user\'s username' do
+      expect(page).to have_content(user1.name)
+    end
+
+    it 'displays the number of posts the user has written' do
+      expect(page).to have_content("Number of posts: #{user1.posts_counter}")
+    end
+
+    it 'displays a post1 title' do
+      expect(page).to have_content(post1.title)
+    end
+
+    it 'displays a post1 text' do
+      expect(page).to have_content(post1.text)
+    end
+
+    it 'displays a post2 title' do
+      expect(page).to have_content(post2.title)
+    end
+
+    it 'displays a post2 text' do
+      expect(page).to have_content(post2.text)
+    end
+
+    it 'displays the number of comments a post has' do
+      expect(page).to have_content("Comments: #{post1.comments_counter}")
+    end
+
+    it 'displays the number of likes a post has' do
+      expect(page).to have_content("Likes: #{post1.likes_counter}")
+    end
+
+    it 'redirects to the post show page when a post is clicked' do
+      click_link post1.title
+      expect(page).to have_current_path(user_post_path(user1, post1))
+      expect(page).to have_content(post1.title)
+    end
+
+    it 'displays the first comments on a post' do
+      within('.card', text: post1.title) do
+        expect(page).to have_content("First Comment: #{comment1.user.name}, #{comment1.text}")
+      end
+    end
```

Comment:
Nice job so far😉. As a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md), kindly add a test that asserts you can see the last 5 comments on a post and a test that asserts you can see a section for pagination if there are more posts than fit on the view.👍🏼. Kindly implement this in your `user_post_index` file👍🏼.

<img width="523" alt="image" src="https://github.com/ReemMohamedAbdelfatah/rails-blog-app/assets/64643995/94706f6d-add3-4225-a130-6a3a5660d45c">

<img width="643" alt="image" src="https://github.com/ReemMohamedAbdelfatah/rails-blog-app/assets/64643995/97448d5b-c1d4-4e49-8e5a-fad2d34534f7">